app/services/alerts/alert_attributes_factory.rb

```diff
@@ -0,0 +1,31 @@
+module Alerts
+  class AlertAttributesFactory
```

Comment:
This could be a Builder rather than a Factory - or even could be added to the Alert Class